lib/iris/fileformats/netcdf/loader.py

```diff
@@ -215,10 +215,41 @@ def _get_cf_var_data(cf_var, filename):
         # See https://github.com/SciTools/iris/issues/4994 "Xarray bridge".
         result = cf_var._data_array
     else:
-        total_bytes = cf_var.size * cf_var.dtype.itemsize
+        # Determine size of data; however can't do this for variable length (VLEN)
+        # arrays as the size of the array can only be known by reading the data.
+        # "Variable length" netCDF types have a datatype of `nc.VLType`.
+        if isinstance(
+            getattr(cf_var, "datatype", None), _thread_safe_nc.netCDF4.VLType
+        ):  # TODO(ChrisB): I am accessing netCDF4 directly here - is this ok? Just for type comparison.
+            # We can't know the size of VLen data without reading the variable from disk
+            # first; see https://github.com/Unidata/netcdf-c/issues/1893
+            msg = (
+                f"netCDF variable `{cf_var.cf_name}` is a variable length type of kind {cf_var.dtype} "
+                "and the total data size cannot be known in advance. This may affect the lazy loading "
+                "of the data."
+            )
+            warnings.warn(msg, category=iris.warnings.IrisLoadWarning)
+
+            # A stab in the dark at the mean size of the variable length arrays (this could be anything):
+            mean_vl_array_len = 10
+
+            # Give user the chance to pass a hint of the average variable length array size via
+            # the chunk control context manager. This allows for better decisions to be made on
+            # whether the data should be lazy-loaded or not.
+            if CHUNK_CONTROL.mode is not CHUNK_CONTROL.Modes.AS_DASK:
+                if chunks := CHUNK_CONTROL.var_dim_chunksizes.get(cf_var.cf_name):
+                    if vl_chunk_hint := chunks.get("_vl_hint"):
```

Comment:
This is quite neat, but I guess it needs documenting **_somewhere_**.
I guess it has to be in the CHUNK_CONTROL docs around [here](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris.fileformats.netcdf.loader.html#iris.fileformats.netcdf.loader.ChunkControl.set) somewhere 
(among the 'dimensions' param, and/or the subsequent Notes section)

Reply:
Yes - since you're happy with this approach I will now put the some docs together.

Reply:
Docstring added to the CHUNK_CONTROL class.
New "Variable-length Datatypes" section added to "Further Topics - NetCDF I/O".